cargo-dylint/tests/dylint.rs

```diff
@@ -329,7 +335,7 @@ fn compare_lines(left: &str, right: &str) {
 // smoelius: Skip examples directory for now.
 fn walkdir(include_examples: bool) -> impl Iterator<Item = walkdir::Result<walkdir::DirEntry>> {
     #[allow(unknown_lints, env_cargo_path)]
-    walkdir::WalkDir::new(Path::new(env!("CARGO_MANIFEST_DIR")).join(".."))
+    walkdir::WalkDir::new(Path::new(current_dir().unwrap().to_str().unwrap()))
```

Comment:
```suggestion
    walkdir::WalkDir::new(".")
```
`WalkDir::new`'s argument must implement [`AsRef<Path>`](https://docs.rs/walkdir/latest/walkdir/struct.WalkDir.html#method.new), which `&str` [does](https://doc.rust-lang.org/std/primitive.str.html#impl-AsRef%3CPath%3E-for-str), so I think this ought to work.

Reply:
This fails the `markdown_link_check` test, as it doesn't create full paths
`Error: ENOENT: no such file or directory, stat \'./cargo-dylint/README.md\'

Reply:
Ah. That is arguably a flaw in the `markdown_link_check` test. But I can fix that in a subsequent PR.